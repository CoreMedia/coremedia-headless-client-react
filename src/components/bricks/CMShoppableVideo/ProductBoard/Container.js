// @flow
// import PropTypes from 'prop-types';
import styled from 'styled-components';

import { media } from '../../../../styles/themes/utils';

const Container = styled.div`
  display: flex;
  flex-wrap: nowrap;
  align-items: center;
  height: 100%;
  transition: transform 0.45s ease-out;
  ${props => media(props.theme.breakpoints.tablet)`
    flex-wrap: wrap;
    flex-direction: row;
    transform: unset;
    justify-content: center;
    align-content: ${props.productboardOverflow ? 'flex-start' : 'center'};
  `};
`;

Container.displayName = 'Container';

/*Container.propTypes = {
  theme: PropTypes.shape({
    breakpoints: PropTypes.shape({
      tablet: PropTypes.number.isRequired,
    }).isRequired,
  }),
};*/

Container.defaultProps = {
  theme: {
    breakpoints: {
      tablet: 768,
    },
  },
};

export default Container;
