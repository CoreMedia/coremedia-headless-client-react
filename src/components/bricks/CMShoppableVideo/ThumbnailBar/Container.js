// @flow
import type { ComponentType } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

type Props = {
  translateX: number,
};

const ThumbnailsContainer: ComponentType<Props> = styled.div`
  transform: translateX(${props => `${props.translateX}px`});
  transition: transform 0.45s ease-out;
`;

ThumbnailsContainer.displayName = 'ThumbnailsContainer';

ThumbnailsContainer.propTypes = {
  translateX: PropTypes.number.isRequired,
};

export default ThumbnailsContainer;
