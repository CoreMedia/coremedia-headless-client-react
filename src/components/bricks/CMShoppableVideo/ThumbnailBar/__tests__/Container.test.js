// @flow
import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import Container from '../Container';

describe('Container Component', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<Container translateX={10}>Content</Container>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
