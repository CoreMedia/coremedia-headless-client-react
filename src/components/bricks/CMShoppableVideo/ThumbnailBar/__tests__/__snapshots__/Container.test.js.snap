// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container Component should render correctly 1`] = `
.c0 {
  -webkit-transform: translateX(10px);
  -ms-transform: translateX(10px);
  transform: translateX(10px);
  -webkit-transition: -webkit-transform 0.45s ease-out;
  -webkit-transition: transform 0.45s ease-out;
  transition: transform 0.45s ease-out;
}

<div
  className="c0"
>
  Content
</div>
`;
